src/api/apiErrors.js

```diff
@@ -61,8 +61,32 @@ export const makeErrorFromApi = (httpStatus: number, data: mixed): Error => {
  * Client errors are often caused by incorrect parameters given to the backend
  * by the client application.
  *
- * A notable difference between a Server (5xx) and Client (4xx) errors is that
- * a client error will not be resolved by waiting and retrying the same request.
+ * A Client (4xx) error will not be resolved by waiting and retrying
+ * the same request.
  */
 export const isClientError = (e: Error): boolean =>
   e instanceof ApiError && e.httpStatus >= 400 && e.httpStatus <= 499;
+
+/**
+ * Is exception caused by a Server Error (5xx)?
+ *
+ * A Server (5xx) error may or may not be resolved by waiting a short
+ * time and retrying the same request.
+ */
+export const isServerError = (e: Error): boolean =>
+  e instanceof ApiError && e.httpStatus >= 500 && e.httpStatus <= 599;
```

Comment:
Hmm. Does this actually happen in practice?

Looking at `makeErrorFromApi` above, which is where we construct `ApiError` objects, it looks like on a typical 5xx response we'll currently throw just an `Error` object, with some text in the message. (We'd throw an `ApiError` on a 5xx response if it comes with a body that's valid JSON and decodes to a nice Zulip API error object `{ result: 'error', msg: string, code?: string | void, ... }`. But a 5xx is a server failure, and may not carry nice well-formed JSON like that.)

So I think we want the code in `src/api/` to be throwing a more distinctive error in that case, and then to have `isServerError` detect that.